uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeaveRequest.php

```diff
@@ -477,6 +477,14 @@ private static function validateEntitlementAndWorkingDayAndBalanceChange($params
 
     $currentBalance = $leavePeriodEntitlement->getBalance();
 
+    if (!empty($params['id'])) {
+      $oldLeaveRequest = self::findById($params['id']);
+
+      if ($oldLeaveRequest && $oldLeaveRequest->id && self::isAlreadyApproved($oldLeaveRequest)) {
```

Comment:
The `findById()` method throws an error if it can't find a record with the given ID. So it either stops there or returns a valid object, meaning there's no need to check if the `$oldLeaveRequest` variable is null